compiler/src/lntoamm/Type.ts

```diff
@@ -153,51 +154,71 @@ export class Interface {
     return true
   }
 
-  static fromAst(interfaceAst: any, scope: Scope) { // TODO: replace ANTLR
+  static fromAst(interfaceAst: LPNode, scope: Scope) {
     // Construct the basic interface, the wrapper type, and insert it into the scope
     // This is all necessary so the interface can self-reference when constructing the function and
     // operator types.
-    const interfacename = interfaceAst.VARNAME(0).getText()
+    const interfacename = interfaceAst.get('variable').t
     let iface = new Interface(interfacename)
     const ifaceType = new Type(interfacename, false, false, {}, {}, null, iface)
     scope.put(interfacename, ifaceType)
 
     // Now, insert the actual declarations of the interface, if there are any (if there are none,
     // it will provide only as much as a type generic -- you can set it to a variable and return it
     // but nothing else, unlike Go's ridiculous interpretation of a bare interface).
-    if (!!interfaceAst.interfacebody() && !!interfaceAst.interfacebody().interfacelist()) {
-      for (const interfaceline of interfaceAst.interfacebody().interfacelist().interfaceline()) {
-        if (!!interfaceline.functiontypeline()) {
-          const functiontypeline = interfaceline.functiontypeline()
-          let functionname = null
-          if (!!functiontypeline.VARNAME()) {
-            functionname = functiontypeline.VARNAME().getText()
-          }
-          const typenames = functiontypeline.functiontype().fulltypename();
-          const returnType = scope.deepGet(typenames[typenames.length - 1].getText()) as Type
+    if (
+      interfaceAst.get('interfacedef').has('interfacebody') &&
+      interfaceAst.get('interfacedef').get('interfacebody').get('interfacelist').has()
+    ) {
+      const interfacelist = interfaceAst
+        .get('interfacedef')
+        .get('interfacebody')
+        .get('interfacelist')
+      const interfacelines = []
+      interfacelines.push(interfacelist.get('interfaceline'))
+      interfacelist.get('cdr').getAll().forEach(l => {
+        interfacelines.push(l.get('interfaceline'))
+      })
```

Comment:
This particular pattern to turn comma/semicolon/etc separated lists into arrays is all over the place. It unfortunately bloats up what was 1 LOC in ANTLR to 5 LOC in LP.

Figuring out a way to restore it in LP would be nice.